### PR TITLE
Eslint: Hooks rules

### PR DESCRIPTION
**Pull request by peterhass**
https://favro.com/organization/aee06c3fe6503c804f507dd9/22c630cc02b09f2d6f98b8b3?card=Goo-33088

### CODE REVIEW

File: javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const useEffect = () => {};
+
+export const useMagicalFriendship = (layoutRef) => (
```
Comment (atomiomi) [inline javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx:3]: test invalid examples?

**Review by atomiomi — CHANGES_REQUESTED** [review]

File: javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx (hunk 1 above)
Comment (peterhass) [reply javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx:3]: I cannot test an invalid example for this rule. It should cover 'ignore effect deps' 

valid: all kind of variables can be used inside an effect without deps check
invalid: ???

**Review by atomiomi — APPROVED** [review]

File: javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx (hunk 1 above)
Comment (suprMax) [reply javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx:3]: i think we could test things like call order of hooks etc. use this for inspiration: https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

Comment (peterhass) [reply javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx:3]: That's a rule we don't touch. We use the defaults from airbnb. I don't think it makes sense to test rules set in airbnb config. I would stick with our own set of rules we apply on top of that

Comment (suprMax) [reply javascript/packages/eslint-config-nebenan/test/valid/hooks.jsx:3]: good point, then good to go as is

**Review by suprMax — APPROVED** [review]

**Pull request merged**